modules/carbonAnalyticsAdapter.js

```diff
@@ -0,0 +1,227 @@
+import { generateUUID } from '../src/utils.js';
+import { ajax } from '../src/ajax.js';
+import { getStorageManager } from '../src/storageManager.js';
+import {getGlobal} from '../src/prebidGlobal.js';
+import adapterManager from '../src/adapterManager.js';
+import adapter from '../libraries/analyticsAdapter/AnalyticsAdapter.js';
+import CONSTANTS from '../src/constants.json';
+
+const CARBON_GVL_ID = 493;
+const ANALYTICS_VERSION = 'v1.0';
+const CARBON_ANALYTICS_URL = 'https://pb-ing.ccgateway.net';
+const PROFILE_ID_KEY = 'carbon_ccuid';
+const PROFILE_ID_COOKIE = 'ccuid';
+const SESSION_ID_COOKIE = 'ccsid';
+const MODULE_NAME = 'carbon';
+const ANALYTICS_TYPE = 'endpoint';
+
+const DAY_MS = 24 * 60 * 60 * 1000;
+const MINUTE_MS = 60 * 1000;
+
+export const storage = getStorageManager({gvlid: CARBON_GVL_ID, moduleName: 'carbon'});
+
+let enabledEvents = [];
+let pageViewId = '';
+let profileId = '';
+let sessionId = '';
+let parentId = '';
+let pageEngagement = {};
+
+let carbonAdapter = Object.assign(adapter({CARBON_ANALYTICS_URL, ANALYTICS_TYPE}), {
+  track({eventType, args}) {
+    if (!enabledEvents.includes(eventType)) return;
```

Comment:
why check for enabled events if we are only ever executing when it is `Auction_End` or `TCF2_ENFORCEMENT` ?

Should we just make those the defaults and remove the ability to set enabledEvents?

Or make some custom generic handler for events so we are able to do more than just the two below?